src/OpenLoco/src/Objects/ObjectUtils.cpp

```diff
@@ -0,0 +1,411 @@
+#include "ObjectUtils.h"
+#include "AirportObject.h"
+#include "BridgeObject.h"
+#include "Date.h"
+#include "DockObject.h"
+#include "ObjectManager.h"
+#include "RoadExtraObject.h"
+#include "RoadObject.h"
+#include "RoadStationObject.h"
+#include "TrackExtraObject.h"
+#include "TrackObject.h"
+#include "TrainSignalObject.h"
+#include "TrainStationObject.h"
+#include "VehicleObject.h"
+#include "World/CompanyManager.h"
+#include <OpenLoco/Core/Numerics.hpp>
+
+namespace OpenLoco
+{
+    // 0x0048D70C
+    sfl::static_vector<uint8_t, 16> getAvailableAirports()
+    {
+        sfl::static_vector<uint8_t, 16> stationList;
+        auto currentYear = getCurrentYear();
+        for (uint8_t i = 0; i < ObjectManager::getMaxObjects(ObjectType::airport); i++)
+        {
+            auto airportObj = ObjectManager::get<AirportObject>(i);
+            if (airportObj == nullptr)
+            {
+                continue;
+            }
+            if (currentYear < airportObj->designedYear)
+            {
+                continue;
+            }
+            if (currentYear > airportObj->obsoleteYear)
+            {
+                continue;
+            }
+            stationList.push_back(i);
+        }
+
+        std::sort(std::begin(stationList), std::end(stationList));
+        return stationList;
+    }
+
+    // 0x0048D753
+    sfl::static_vector<uint8_t, 16> getAvailableDocks()
+    {
+        sfl::static_vector<uint8_t, 16> stationList;
+        auto currentYear = getCurrentYear();
+        for (uint8_t i = 0; i < ObjectManager::getMaxObjects(ObjectType::dock); i++)
```

Comment:
getMaxObjects would return 8 for ObjectType::dock but the static vector has 16 as capacity, I'm not sure where the number 16 comes from but we should probably have actual constants for this.

Reply:
The 16 comes from the fact it uses the same memory address to store the result for all the station variations.